bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/capability/SecurityCapability.java

```diff
@@ -0,0 +1,180 @@
+            }
+            HomeDataModule data = cameras.get(handler.getId());
+            if (data != null) {
+                handler.setNewData(data);
+            }
+        });
+    }
+
+    @Override
+    protected void updateHomeStatus(HomeStatus homeStatus) {
+        NAObjectMap<HomeStatusPerson> persons = homeStatus.getPersons();
+        NAObjectMap<HomeStatusModule> cameras = homeStatus.getModules();
+        for (CommonInterface handler : handler.getActiveChildren()) {
+            HomeStatusPerson dataPerson = persons.get(handler.getId());
+            if (dataPerson != null) {
+                handler.setNewData(dataPerson);
+            }
+            HomeStatusModule dataCamera = cameras.get(handler.getId());
+            if (dataCamera != null) {
+                handler.setNewData(dataCamera);
+            }
+        }
+    }
+
+    @Override
+    protected void updateHomeEvent(HomeEvent homeEvent) {
+        String personId = homeEvent.getPersonId();
+        if (personId != null) {
+            handler.getActiveChildren().stream().filter(handler -> personId.equals(handler.getId())).findFirst()
+                    .ifPresent(handler -> handler.setNewData(homeEvent));
+        }
+        String cameraId = homeEvent.getCameraId();
+        handler.getActiveChildren().stream().filter(handler -> cameraId.equals(handler.getId())).findFirst()
+                .ifPresent(handler -> handler.setNewData(homeEvent));
+    }
+
+    Collection<HomeEvent> getCameraEvents(String cameraId) {
+        return api.map(api -> {
+            try {
+                return api.getCameraEvents(handler.getId(), cameraId);
+            } catch (NetatmoException | NoSuchElementException e) {
+                logger.warn("Error retrieving last events of camera '{}' : {}", cameraId, e.getMessage());
+                return null;
+            }
+        }).orElse(List.of());
+    }
+
+    Collection<HomeEvent> getPersonEvents(String personId) {
+        return api.map(api -> {
+            try {
+                return api.getPersonEvents(handler.getId(), personId);
+            } catch (NetatmoException | NoSuchElementException e) {
+                logger.warn("Error retrieving last events of person '{}' : {}", personId, e.getMessage());
+                return null;
+            }
+        }).orElse(List.of());
+    }
+
+    void setPersonAway(String personId, boolean away) {
+        api.ifPresent(api -> {
+            try {
+                api.setPersonAwayStatus(handler.getId(), personId, away);
+                handler.expireData();
+            } catch (NetatmoException | NoSuchElementException e) {
+                logger.warn("Error setting person away/at home '{}' : {}", personId, e.getMessage());
+            }
+        });
+    }
+
+    @Nullable
+    String ping(String vpnUrl) {
+        return api.map(api -> {
+            try {
+                return api.ping(vpnUrl);
+            } catch (NetatmoException | NoSuchElementException e) {
```

Comment:
NoSuchElementException can be thrown by what exactly ?

Reply:
Left over, corrected.